diff --git a/build.go b/build.go
index 7d400d6..d0b246f 100644
--- a/build.go
+++ b/build.go
@@ -434,10 +434,15 @@ func build(target target, tags []string) {
 	lazyRebuildAssets()
 
 	tags = append(target.tags, tags...)
+	tags = append(tags, "rpm_crashtraceback")
 
 	rmr(target.BinaryName())
 
-	args := []string{"build", "-i", "-v"}
+	build_id, _ := ioutil.ReadFile("build_id")
+	fedora_ldflags := fmt.Sprintf("%s %s -B 0x%s -extldflags '-Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld'",
+		ldflags(), os.Getenv("LDFLAGS"), build_id)
+
+	args := []string{"build", "-compiler", "gc", "-ldflags", fedora_ldflags, "-a", "-v", "-x"}
 	args = appendParameters(args, tags, target)
 
 	os.Setenv("GOOS", goos)
@@ -736,11 +741,10 @@ func ldflags() string {
 	}
 
 	b := new(bytes.Buffer)
-	b.WriteString("-w")
-	fmt.Fprintf(b, " -X main.Version%c%s", sep, version)
-	fmt.Fprintf(b, " -X main.BuildStamp%c%d", sep, buildStamp())
-	fmt.Fprintf(b, " -X main.BuildUser%c%s", sep, buildUser())
-	fmt.Fprintf(b, " -X main.BuildHost%c%s", sep, buildHost())
+	fmt.Fprintf(b, "-X main.Version%c%s ", sep, version)
+	fmt.Fprintf(b, "-X main.BuildStamp%c%d ", sep, buildStamp())
+	fmt.Fprintf(b, "-X main.BuildUser%c%s ", sep, buildUser())
+	fmt.Fprintf(b, "-X main.BuildHost%c%s", sep, buildHost())
 	return b.String()
 }
 
