From 8d745e076d52612b346209828ca81cc0fc026c3b Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Sun, 30 Sep 2018 18:07:55 +0200
Subject: [PATCH] lib/scanner/blocks_test.go: adapt to rollinghash changes

---
 lib/scanner/blocks_test.go | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/scanner/blocks_test.go b/lib/scanner/blocks_test.go
index 311dbdd..99f1a71 100644
--- a/lib/scanner/blocks_test.go
+++ b/lib/scanner/blocks_test.go
@@ -144,14 +144,13 @@ func TestAdler32Variants(t *testing.T) {
 
 	windowSize := 128
 
-	hf2.Reset()
-
 	hf3 := rollingAdler32.New()
 	hf3.Write(data[:windowSize])
 
 	for i := windowSize; i < len(data); i++ {
 		if i%windowSize == 0 {
 			// let the reference function catch up
+			hf2.Reset()
 			hf2.Write(data[i-windowSize : i])
 
 			// verify that they are in sync with the rolling function
-- 
2.17.1

