From 3d8344003ef7724a00e37f59d3fd5fa2b13973a1 Mon Sep 17 00:00:00 2001
From: Jakob Borg <jakob@kastelo.net>
Date: Thu, 19 Jul 2018 20:49:05 +0200
Subject: [PATCH] lib/logger: Add missing dots (fixes #5073)

---
 lib/logger/logger.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/logger/logger.go b/lib/logger/logger.go
index e5656ff7e..e4c62b49b 100644
--- a/lib/logger/logger.go
+++ b/lib/logger/logger.go
@@ -106,7 +106,7 @@ func (l *logger) callHandlers(level LogLevel, s string) {
 
 // Debugln logs a line with a DEBUG prefix.
 func (l *logger) Debugln(vals ...interface{}) {
-	l.debugln(3, vals)
+	l.debugln(3, vals...)
 }
 func (l *logger) debugln(level int, vals ...interface{}) {
 	s := fmt.Sprintln(vals...)
